sys/crypto/modes/ccm.c

```diff
@@ -76,7 +82,9 @@ static int ccm_create_mac_iv(cipher_t *cipher, uint8_t auth_data_len, uint8_t M,
     /* set flags in B[0] - bit format:
             7        6     5..3  2..0
         Reserved   Adata    M_    L_    */
-    M_ = (M - 2) / 2;
+    /* valid M values include 0 for CCMS */
+    M_ = M == 0 ? 0 : (M - 2) / 2;
```

Comment:
I see that `ccm_create_mac_iv` is just a static function. But anyway, before this change `M_` had a different result `if M=0`

Reply:
According to [Section 2.3 Point 4.](https://mentor.ieee.org/802.15/dcn/04/15-04-0537-02-004b-formal-specification-ccm-star-mode-operation.pdf):

"The value M=0 corresponds to disabling authenticity, since then the authentication field is the empty string"

-> does it need further action?

Reply:
No, I think I didnt't like the `M-2` when M==0, but I can leave it as it was before, or I can return right away if `M=0`.

Reply:
> I see that `ccm_create_mac_iv` is just a static function. But anyway, before this change `M_` had a different result `if M=0`

Before it would have been invalid for `M` to be `0`, its not a valid input for `ccm`, and it is checked through `_valid_mac_length_ccm`.

Reply:
Skipped it entirely in 1f2698c if `mac_length=0`.